internal/provider-config/config.go

```diff
@@ -428,9 +428,11 @@ func (p *LxdProviderConfig) setRemote(remoteName string, remote LxdRemote) error
 
 // acceptRemoteCertificate retrieves the unverified peer certificate found at
 // the remote address and stores it locally.
-func (p *LxdProviderConfig) acceptRemoteCertificate(remoteName string, url string) error {
-	// Check if we are allowed to accept the remote certificate.
-	if !p.acceptServerCertificate {
+func (p *LxdProviderConfig) acceptRemoteCertificate(remoteName string, token string, url string) error {
+	// Check if we are allowed to blindly accept the remote certificate.
+	// When the trust token is used, the fingerprint contained in the token
+	// is used to ensure we get the right certificate.
+	if token == "" && !p.acceptServerCertificate {
 		return errors.New("Unable to communicate with remote server. " +
 			`Either set "accept_remote_certificate" to true or add ` +
 			"the remote out of band of Terraform and try again.")
```

Comment:
Here again we should recommend using a remote add token as well.

Reply:
Yep, please check if it is ok now.